File: waybar/timer.py
#pyright: reportUnusedCallResult=false
import sys
import time
import os
home = os.getenv("HOME") or "/"

if "targetTime.json" not in os.listdir(home+"/dotfiles/waybar/"):
    with open(home+"/dotfiles/waybar/targetTime.json","w") as f:
        f.write("0")  
while True:
    with open(home+"/dotfiles/waybar/targetTime.json") as f:
        target=int(float(f.read())-time.time())


    if target>0:
        hour=target//3600
        minute=(target%3600)//60
        second=target%60
        formatted=f'{hour:02}:{minute:02}:{second:02}\n'.replace("00:","",1)
        sys.stdout.write(formatted)
    else:
        sys.stdout.write("Timer\n")
    sys.stdout.flush()
    time.sleep(1)

File: waybar/timer.cpp
#include <unistd.h>
#include <cmath>
#include <sys/types.h>
#include <pwd.h>
#include <iostream>
#include <fstream>
#include <string>
#include <chrono>
#include <thread>
#include <unistd.h>
using namespace std;

string target;
int targetTime;

int getRemaining() {
    auto curtimeclock = chrono::system_clock::now();
    int curtime = chrono::duration_cast<chrono::seconds>(curtimeclock.time_since_epoch()).count();
    int targetdiff = targetTime - curtime;
    return targetdiff;
}

int main() {
    struct passwd *pw = getpwuid(getuid());
    string homedir = pw->pw_dir;
    
    
    while (true) {
        ifstream inFile;
        inFile.open(homedir + "/dotfiles/waybar/targetTime.json");
        getline (inFile,target);
        inFile.close();
        targetTime = stoi(target);
        
        int diff = getRemaining();
        
        if (diff < 0) {
            cout << "Timer\n";
        } else {
            int hour = floor(diff / 3600);
            int minute = floor((diff % 3600)/60);
            int seconds = diff % 60;
            
            if (hour <= 0) {
                cout << minute << ":" << seconds << "\n";
            } else {
                cout << hour << ":" << minute << ":" << seconds << "\n";
            }
        }
        cout.flush();
        sleep(1);
    }
}
